second-hand-b/src/main/java/com/codesquad/secondhand/adapter/in/web/config/jwt/JwtSignInAuthenticationFilter.java

```diff
@@ -0,0 +1,53 @@
+package com.codesquad.secondhand.adapter.in.web.config.jwt;
+
+import com.codesquad.secondhand.adapter.in.web.security.JwtAuthenticationToken;
+import com.codesquad.secondhand.application.port.out.MemberRepository;
+import com.codesquad.secondhand.domain.member.Member;
+import com.codesquad.secondhand.domain.units.JwtTokenProvider;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+public class JwtSignInAuthenticationFilter extends OncePerRequestFilter {
+
+    private final JwtTokenProvider jwtTokenProvider;
+    private final MemberRepository memberRepository;
+
+    public JwtSignInAuthenticationFilter(JwtTokenProvider jwtTokenProvider, MemberRepository memberRepository) {
+        this.jwtTokenProvider = jwtTokenProvider;
+        this.memberRepository = memberRepository;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+            FilterChain filterChain) throws ServletException, IOException {
+        Authentication existsAuthentication = SecurityContextHolder.getContext().getAuthentication();
+        if (existsAuthentication != null && existsAuthentication.isAuthenticated()) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+        String token = jwtTokenProvider.resolveToken(request);
+        if (token != null && jwtTokenProvider.validateToken(token.split(" ")[1].trim())
```

Comment:
이건 Java 가 Nullable 언어가 아니여서 고려해보시면 좋은 점인데요. 
보통 아래에서 null 체크를 하고 로직안에서 Null 임을 확실하게 하려면 final 을 붙여서 재할당을 막는 방법도 있습니다.
```suggestion
        final String token = jwtTokenProvider.resolveToken(request);
        if (token != null && jwtTokenProvider.validateToken(token.split(" ")[1].trim())
```